packages/blockchain-wallet-v4-frontend/src/scenes/Settings/Addresses/Btc/ManageAddresses/UsedAddresses/index.js

```diff
@@ -21,13 +22,25 @@ class UsedAddressesContainer extends React.PureComponent {
   }
 
   render () {
-    return <UsedAddressesTemplate usedAddressesVisible={this.props.usedAddressesVisible} onShowUsedAddresses={this.onShowUsedAddresses} />
+    const { usedAddressesVisible, walletIndex } = this.props
+
+    return <UsedAddressesShowTemplate usedAddressesVisible={usedAddressesVisible} onShowUsedAddresses={this.onShowUsedAddresses} walletIndex={walletIndex} />
   }
 }
 
-const mapStateToProps = (state, ownProps) => ({
-  usedAddressesVisible: selectors.components.usedAddresses.getWalletUsedAddressVisibility(state, ownProps.walletIndex)
-})
+const mapStateToProps = (state, ownProps) => {
+  // since usedAddresses state tree is stored based on walletIndexes, the default states will not exist
+  // on first attempt to access
+  if (!path(['state', 'components', 'usedAddresses', ownProps.walletIndex])) {
```

Comment:
I think if you address the comment in reducer you can delete this

Reply:
I just realized this is keyed by walletIndex, so my suggestion in reducer wont' work, but I still really don't like this, setting the state anywhere other than reducers is a big no.

Reply:
This is related to comment above, but now in the component I know the current walletIndex and can set an initial state if it hasnt been done so already.